pkgs/applications/virtualization/libgovirt/default.nix

```diff
@@ -1,46 +1,48 @@
 { lib
 , stdenv
-, fetchurl
-, glib
+, fetchzip
 , gnome
-, librest
-, libsoup
+, meson
 , pkg-config
 , gobject-introspection
+, ninja
+, glib
+, librest_1_0
 }:
 
 stdenv.mkDerivation rec {
   pname = "libgovirt";
-  version = "0.3.8";
+  version = "0.3.9";
 
   outputs = [ "out" "dev" ];
 
-  src = fetchurl {
+  src = fetchzip {
```

Comment:
Is there a reason to switch to `fetchzip`?

Reply:
As mentioned, I prefer to have the source files directly. It's easier to introspect `libgovirt.src` if it's just a regular tree of files in an output path rather than a file in some arbitrary archive format you need to unpack somewhere first. Fetchzip handles all of that internally.

It also enables some optimisations for users of modern filesystems as source files are cheaply reflinked rather than copied/unpacked.

I also generally prefer to use the most specialised tool for any given task. You could `fetchurl` a tarball off github and that'd work just fine but I'd always prefer to use `fetchFromGitHub` instead. It's the tool you use for Github sources. `fetchzip` is the tool to fetch archives.

Reply:
Fair. One of my concerns was that `fetchzip` would pull `unzip` as an unnecessary dependency, but your arguments look good to me.

Reply:
Note that all of this is about build-time dependencies. Anyone who's regularly building libraries should be prepared for large build-time closures for pretty much anything and unzip is a drop in the water in that case.

Reply:
See https://discourse.nixos.org/t/github-archive-checksums-may-change/25108 for another reason to prefer specialised fetchers.

Reply:
I don't expect this to happen from time to time, as other distros may rely on the determinism of the archive itself. Also, GitHub has [reverted the change](https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/).

Meanwhile, this did happen on one of my packages (ocamlPackages.lustre-v6). It might be beneficial to switch to `fetchzip` there.

Reply:
These are static files uploaded to FTP. There is no chance of them changing.